elements/rh-pagination/rh-pagination.ts

```diff
@@ -0,0 +1,179 @@
+  </svg>`;
+
+const L2 = html`
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 17.44 14">
+    <path d="M8.7 6.26 14.66.3a1.05 1.05 0 0 1 1.49 0l.98.99c.42.4.42 1.07 0 1.48L12.92 7l4.2 4.22c.42.4.42 1.07 0 1.48l-.98 1c-.41.4-1.08.4-1.48 0L8.7 7.73a1.05 1.05 0 0 1 0-1.48zM.3 7.74l5.96 5.95c.4.41 1.07.41 1.48 0l.99-.99c.4-.4.4-1.07 0-1.48L4.52 7l4.21-4.22c.41-.4.41-1.07 0-1.48l-.99-1a1.05 1.05 0 0 0-1.48 0L.31 6.27a1.05 1.05 0 0 0 0 1.48z"/>
+  </svg>`;
+
+/**
+ * Pagination
+ * @slot - Place element content here
+ */
+@customElement('rh-pagination')
+export class RhPagination extends LitElement {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [styles];
+
+  /**
+   * Override `overflow` values set from HTML or JS.
+   * `overflow` should ideally be private, but because
+   * we can't do `::slotted(nav ol li)`, we need to reflect
+   * it to a host attribute, so that lightdom CSS can target
+   * the list items.
+   */
+  @property({ reflect: true }) overflow: 'start' | 'end' | 'both' | null = null;
+
+  /** Accessible label for the 'first page' button */
+  @property({ attribute: 'label-first' }) labelFirst = 'first page';
+  /** Accessible label for the 'previous page' button */
+  @property({ attribute: 'label-previous' }) labelPrevious = 'previous page';
+  /** Accessible label for the 'next page' button */
+  @property({ attribute: 'label-next' }) labelNext = 'next page';
+  /** Accessible label for the 'last page' button */
+  @property({ attribute: 'label-last' }) labelLast = 'last page';
+
+  #screenSize = new ScreenSizeController(this);
+  #logger = new Logger(this);
+
+  #nav = this.querySelector('nav');
+  #links = this.#nav?.querySelectorAll<HTMLAnchorElement>('li a');
+  #currentLink = this.#getCurrentLink();
+
+  #currentIndex = -1;
+  #firstLink: HTMLAnchorElement | null = null;
+  #lastLink: HTMLAnchorElement | null = null;
+  #nextLink: HTMLAnchorElement | null = null;
+  #prevLink: HTMLAnchorElement | null = null;
+
+  update(changed: PropertyValues<this>): void {
+    this.#update();
+    super.update(changed);
+  }
+
+  render() {
+    const { mobile, size } = this.#screenSize;
+    const { labelFirst, labelPrevious, labelNext, labelLast } = this;
+    const firstHref = this.#currentLink === this.#firstLink ? undefined : this.#firstLink?.href;
+    const prevHref = this.#prevLink?.href;
+    const nextHref = this.#nextLink?.href;
+    const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
+    return html`
+      <div id="container" class=${classMap({ mobile, [size as string]: true })}>
+        <a id="first" class="stepper" href=${ifDefined(firstHref)} aria-label=${labelFirst}>${L2}</a>
+        <a id="prev" class="stepper" href=${ifDefined(prevHref)} aria-label=${labelPrevious}>${L1}</a>
+
+        <div id="nav-container" ?hidden=${mobile}>
+          <slot @slotchange=${this.#update}></slot>
```

Comment:
We've struggled with this is the past with PFE.  Should we use slotchange or mutation observer with a child tree.  We opted for mutation observers to make it easiest on the our users.

Example, this wont fire the "#update()" function

```
for (let item of [6,7,8,9]) {
    document.querySelector('rh-pagination ol li:last-of-type').insertAdjacentHTML('afterend', `<li><a>${item}</a></li>`)
}
```

Reply:
agreed. fixed